backend/src/utils/openapi.py

```
@@ -0,0 +1,36 @@
+	Args:
+		url (str): URL to fetch OpenAPI spec from. Defaults to localhost:8080.
+		
+	Returns:
+		dict: OpenAPI specification as a dictionary
+	"""
+	try:
+		response = httpx.get(url)
+		response.raise_for_status()
+		return response.json()
+	except httpx.HTTPError as e:
+		raise Exception(f"Failed to fetch OpenAPI spec: {str(e)}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)